### PR TITLE
docs: Add PostgreSQL CDC to Snowflake tutorial.

### PR DESCRIPTION
**Pull request by danthelion**
**Description:**

Add tutorial to showcase how to set up a PostgreSQL CDC to Snowflake data flow.

**Documentation links affected:**

- + New doc page for the tutorial

- - -
This change is [Reviewable](https://reviewable.io/reviews/estuary/flow/1452)

### CODE REVIEW

**Comment by danthelion** [comment]
@travjenkins That's a fair concern, I'll stay on top of keeping them in sync for now

**Review by jonwihl — APPROVED** [review]
LGTM

File: site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+CREATE USER flow_capture WITH PASSWORD 'secret' REPLICATION;
+```
+
+This user is dedicated to the CDC replication process. It is granted the necessary permissions to read all data from the database, allowing it to capture changes across tables efficiently. In a production environment, make sure you use a more secure password than what is in the example.
+
+```sql
+GRANT pg_read_all_data TO flow_capture;
+```
+
+Granting the `pg_read_all_data` privilege to the `flow_capture` user ensures that it can access and read data from all tables in the database, essential for capturing changes.
```
Comment (willdonnelly) [inline site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md:165]: It might be useful to mention here that `pg_read_all_data` is used for convenience in the tutorial but is not a hard requirement, since it is possible to grant a more restricted set of permissions and our docs discuss that in more detail.

File: site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+
+```sql
+CREATE TABLE products (...)
+```
+
+The source tables, such as the `products` table in this example, contain the data whose changes we want to capture and replicate. These tables must have a primary key defined to uniquely identify rows.
```
Comment (willdonnelly) [inline site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md:171]: A primary key is not necessarily required and implying that it is could turn some users off of our product, so I'd recommend either dropping that sentence or changing it to make it more clear that it's just a recommendation.

File: site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+
+```sql
+CREATE PUBLICATION flow_publication;
+ALTER PUBLICATION flow_publication SET (publish_via_partition_root = true);
+ALTER PUBLICATION flow_publication ADD TABLE public.flow_watermarks, public.products;
+```
+
+A publication defines a set of tables whose changes will be replicated. In this case, the `flow_publication` publication includes the `public.flow_watermarks` and `public.products` tables.
+
+These commands configure the `flow_publication` publication to publish changes via partition root and add the specified tables to the publication. By setting `publish_via_partition_root` to true, the publication ensures that updates to partitioned tables are correctly captured and replicated.
```
Comment (willdonnelly) [inline site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md:193]: Might add a note "The table in this tutorial is not partitioned, but we recommend always setting `publish_via_partition_root` when creating a publication"

**Review by willdonnelly — COMMENTED** [review]
I meant to skim through this sooner and offer some feedback on the content. I'm a bit late to the party but I think a couple of the comments are still relevant, most notably I'm pretty sure the Snowflake capture needs to have its update delay overridden to 0 or 60s or something if we want to show off how low-latency we are.

File: site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md (hunk 1 above)
Comment (danthelion) [reply site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md:165]: Thanks, that's a good flag!

File: site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md (hunk 2 above)
Comment (danthelion) [reply site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md:171]: Thanks, will update this section

File: site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+CREATE TABLE products (...)
+```
+
+The source tables, such as the `products` table in this example, contain the data whose changes we want to capture and replicate. These tables must have a primary key defined to uniquely identify rows.
+
+```sql
+CREATE TABLE IF NOT EXISTS public.flow_watermarks (...)
+```
+
+The `flow_watermarks` table is used to store metadata about the replication process, specifically the position of the replication slot and the corresponding watermark. This information helps track the progress of replication and ensures consistency.
```
[the remark this answers is not in the document]
Comment (danthelion) [reply site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md:177]: Makes sense, is the watermarking strategy generalized across connectors? It would be useful to have a separate "concept guide" around it. Either way, I'll update this section in this tutorial

File: site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md (hunk 3 above)
Comment (danthelion) [reply site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md:193]: 👍

File: site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+(1 row)
+
+postgres=# select count(*) from products;
+ count
+-------
+  2642
+(1 row)
+```
+
+By executing a `count(*)` statement a few seconds apart you are able to verify that data is in fact being replicated into the table.
```
[the remark this answers is not in the document]
Comment (danthelion) [reply site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md:256]: You're right, thanks!

File: site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+
+A successful deployment will look something like this:
+
+![Successful Deployment screen](https://storage.googleapis.com/estuary-marketing-strapi-uploads/uploads//publish_successful_4e18642288/publish_successful_4e18642288.png)
+
+And that’s pretty much it, you’ve successfully published a real-time CDC pipeline. Let’s check out Snowflake to see how the data looks.
+
+![Results in Snowflake](https://storage.googleapis.com/estuary-marketing-strapi-uploads/uploads//snowflake_verification_2eb047efec/snowflake_verification_2eb047efec.png)
+
+Looks like the data is arriving as expected, and the schema of the table is properly configured by the connector based on the types of the original table in Postgres. To get a feel for how low latency the pipeline is, feel free to jump back into the terminal and insert or modify some records in our products table then head over to either the Flow web UI or straight to Snowflake and query the materialized dataset to see the changes!
```
[the remark this answers is not in the document]
Comment (danthelion) [reply site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md:393]: You're right, I used low-latency a bit too liberally here, I'll add the notes about the update delay and rephrase it to be more correct.
